### PR TITLE
Merge `curl` options used by casks into download strategies.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Add `curl` options from cask download strategy to `CurlDownloadStrategy`.

### CODE REVIEW

File: Library/Homebrew/download_strategy.rb [hunk 1]
```diff
@@ -323,7 +342,13 @@ def _fetch
 # Query parameters on the URL are converted into POST parameters
 class CurlPostDownloadStrategy < CurlDownloadStrategy
   def _fetch
-    base_url, data = @url.split("?")
+    base_url, data = if meta.key?(:data)
+      escape_data = ->(k, v) { ["-d", "#{CGI.escape(k.to_s)}=#{CGI.escape(v.to_s)}"] }
```
Comment (MikeMcQuaid) [inline Library/Homebrew/download_strategy.rb]: Can the `"#{CGI.escape(k.to_s)}=#{CGI.escape(v.to_s)}"` bit maybe be a shared function where it's used above too?

Comment (reitermarkus) [reply Library/Homebrew/download_strategy.rb]: I'd say let's go with the rule of three on this one.

Comment (reitermarkus) [reply Library/Homebrew/download_strategy.rb]: Made me think, that there should be something like this for `URI`. And there is: `URI.encode_www_form`

Comment (MikeMcQuaid) [reply Library/Homebrew/download_strategy.rb]: Nice!

**Pull request merged**